include/heyoka/mascon.hpp

```diff
@@ -0,0 +1,212 @@
+            }
+        }();
+
+        // mascon_points (no default)
+        std::vector<std::vector<number>> mascon_points;
+        if constexpr (p.has(kw::mascon_points)) {
+            for (const auto &point : p(kw::mascon_points)) {
+                mascon_points.emplace_back(std::vector<number>{number{point[0]}, number{point[1]}, number{point[2]}});
+            }
+        } else {
+            throw std::invalid_argument("mascon_points is missing from the kwarg list!");
+        };
+
+        // mascon_masses (no default)
+        std::vector<number> mascon_masses;
+        if constexpr (p.has(kw::mascon_masses)) {
+            for (const auto &mass : p(kw::mascon_masses)) {
+                mascon_masses.emplace_back(mass);
+            }
+        } else {
+            throw std::invalid_argument("mascon_masses is missing from the kwarg list!");
+        };
+
+        // omega (no default)
+        number pn(0.), qn(0.), rn(0.);
+        if constexpr (p.has(kw::omega)) {
+            pn = number{p(kw::omega)[0]};
+            qn = number{p(kw::omega)[1]};
+            rn = number{p(kw::omega)[2]};
+        } else {
+            throw std::invalid_argument("omega is missing from the kwarg list!");
+        };
+
+        // 3 - Create the return value.
+        std::vector<std::pair<expression, expression>> retval;
+        // 4 - Main code
+        auto dim = std::size(mascon_masses);
+        auto [x, y, z, vx, vy, vz] = make_vars("x", "y", "z", "vx", "vy", "vz");
+        // Assemble the contributions to the x/y/z accelerations from each mass.
+        std::vector<expression> x_acc, y_acc, z_acc;
+        // Assembling the r.h.s.
+        // FIRST: the acceleration due to the mascon points
+        for (decltype(dim) i = 0; i < dim; ++i) {
+            auto x_masc = expression{mascon_points[i][0]};
+            auto y_masc = expression{mascon_points[i][1]};
+            auto z_masc = expression{mascon_points[i][2]};
+            auto m_masc = expression{mascon_masses[i]};
+            auto xdiff = (x - x_masc);
+            auto ydiff = (y - y_masc);
+            auto zdiff = (z - z_masc);
+            auto r2 = square(xdiff) + square(ydiff) + square(zdiff);
+            auto common_factor = -Gconst * m_masc * pow(r2, expression{number{-3. / 2.}});
+            x_acc.push_back(common_factor * xdiff);
+            y_acc.push_back(common_factor * ydiff);
+            z_acc.push_back(common_factor * zdiff);
+        }
+        // SECOND: centripetal and Coriolis
+        auto p = expression{pn};
+        auto q = expression{qn};
+        auto r = expression{rn};
+        // w x w x r
+        auto centripetal_x = -q * q * x - r * r * x + q * y * p + r * z * p;
+        auto centripetal_y = -p * p * y - r * r * y + p * x * q + r * z * q;
+        auto centripetal_z = -p * p * z - q * q * z + p * x * r + q * y * r;
+        // 2 w x v
+        auto coriolis_x = expression{number{2.}} * (q * vz - r * vy);
```

Comment:
There are new constructors for expression directly from double/long double/etc. if you want to save some typing.